src/arch/svm/SpokeUtils.ts

```diff
@@ -150,6 +173,22 @@ export function fillStatusArray(
   throw new Error("fillStatusArray: not implemented");
 }
 
+/**
+ * Returns the PDA for the State account.
+ * @param programId The SpokePool program ID.
+ * @param extraSeed An optional extra seed. Defaults to 0.
+ * @returns The PDA for the State account.
+ */
+export async function getStatePda(programId: string, extraSeed = 0): Promise<Address> {
```

Comment:
I generally just copied the format for these functions from https://github.com/across-protocol/sdk/pull/978